### PR TITLE
Downloadable Tables

### LINKED ISSUE #46: Make interactive tables
Make tables which can be sorted by any column and exported.  

Similar to the exomast implementation:
https://catalogs.mast.stsci.edu/eaot

### PR DESCRIPTION
**Pull request by Will-Cooper**
Makes every returned table returnable.
Can batch download spectra.
For queries which return multiple table, batch them and download.
Closes #46.

### CODE REVIEW

**Comment by Will-Cooper** [comment]
```python
response = make_response(file_as_string)
response.headers["Content-Disposition"] = "attachment; filename=%s" % filename 
```

**Comment by dr-rodriguez** [comment]
Noticing that on https://status.heroku.com/ there are remarks of a security incident at Heroku and they've disabled Heroku-GitHub integration for the time being. Just in case you've had issues deploying your updates. I also wonder if this could impact your attempt at downloads (I suspect not, but good to keep in mind).

File: simple_app/app_simple.py [hunk 1]
```diff
@@ -171,6 +187,141 @@ def autocomplete():
...
+        The dataframe string
+    """
+    query = curdoc().template_variables['query']
+    db = SimpleDB(db_file, connection_arguments={'check_same_thread': False})  # open database
+    resultdict: dict = db.inventory(query)  # get everything about that object
+    everything = Inventory(resultdict, args, rtnmk=False)
+    if key in resultdict:
+        results: pd.DataFrame = getattr(everything, key.lower())
+        fname = write_file(results).split('/')[-1]
+        return redirect(url_for('download_file', filename=fname))
```
Comment (dr-rodriguez) [inline simple_app/app_simple.py:207]: Why do we have to generate a file first to send it to the user? Can we not send the data directly? Something along the lines of what is described in https://flask.palletsprojects.com/en/1.0.x/patterns/streaming/

File: simple_app/app_simple.py [hunk 2]
```diff
@@ -171,6 +187,141 @@ def autocomplete():
...
+    return None
+
+
+@app_simple.route('/write_spectra', methods=['GET', 'POST'])
+def create_spectrafile_for_download():
+    """
+    Downloads the spectra files and zips together
```
Comment (dr-rodriguez) [inline simple_app/app_simple.py:214]: At MAST, this approach has caused issues from time to time. The logic being that you are downloading from the cloud to a local disk, then bundling it and sending it back. It is far better to just point to use to the individual files and have them get it from the source. Which means no bundling of files, or at best, generating a script for them to download on their end.

**Comment by dr-rodriguez** [comment]
I see what you are trying to do with the /tmp directory, but we really shouldn't need to store local files to send them to the user. CSV tables can be generated on the fly in memory and transmitted without going to the file system. Spectra should not be downloaded to then be uploaded again, at least in my opinion.

**Comment by dr-rodriguez** [comment]
Related to #103 and #104

**Review by dr-rodriguez — APPROVED** [review]
Looks OK. We'll want to address the other issues at some future point.

**Pull request merged**